### DISCUSSION TITLE
Postblock: Unblocking UI suggestions

### DISCUSSION THREAD

**Discussion by marcustyphoon** [AprilSylph/XKit-Rewritten#402] [Suggestion Box]
Very minor UI suggestion: the area displaying posts that can be unblocked could have a max height and scroll bar if it overflows, in case people have many, many blocked posts. (To be fair, I would totally understand if people thought this was actually worse - nested scroll bars are admittedly not the greatest thing ever. Figured I'd mention it, though.)

Maybe-not-worth-it suggestion: blocking a post could save its URL, thus allowing the unblock UI to have some indication what the post actually is (hyperlink, presumably). The storage doesn't seem like much of an issue. Sure, going though and clicking all of the posts you have blocked doesn't sound particularly healthy, but if the unblocking UI is going to be useful at all, it may as well be useful enough to find what post you're trying to unblock.

**Comment by AprilSylph** [comment]
> To be fair, I would totally understand if people thought this was actually worse - nested scroll bars are admittedly not the greatest thing ever.

Yeah, I find nested scrollbars kinda annoying. It does make me think about making the entire preferences interface closer to how I originally envisioned it - with opening a feature's preferences making the feature take over the entire section - but that's going to be a lot more work, and may not be worth doing.

> blocking a post could save its URL, thus allowing the unblock UI to have some indication what the post actually is (hyperlink, presumably)

I would love to do this, but those links would rot very easily. Tumblr needs a way for UUIDs to be translated into blog URLs without an authorised API call tbqh. I think I tested at some point that `shortUrl` links break across blog name changes, but maybe I'm wrong/misremembering/outdated. Something to test again for sure.

**Reply by marcustyphoon** [reply]
Ah, right, URL changes. Yep, with you on both points!

**Comment by marcustyphoon** [comment]
I think we should be able to do this now that /blog/view can take UUIDs, right?
